added a troubleshooting section for macOS mojave (#312)
crystal-lang/crystal-book: installation/on_mac_osx_using_homebrew.md

$ xcode-select --install
$ xcode-select --switch /Library/Developer/CommandLineTools
```

> ## Troubleshooting on macOS MacOS 10.14.2 (Mojave)

Reviewer: ```suggestion
## Troubleshooting on macOS 10.14.2 (Mojave)
```

> if you get an error like:

Reviewer: ```suggestion
If you get an error like:
```

```
ld: library not found for -lssl (this usually means you need to install the development package for libssl)
```

you may need to install OpenSSL and link pkg-config to OpenSSL

```
brew install openssl
```

```
export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig

> ```

Reviewer: I would append something like:

> As with other keg-only formulas there are some caveats shown in `brew info <formula>` that advertise how to link pkg-config with this library.
> 
> Crystal compiler by default will use pkg-config to find the location of libraries to link.

$ xcode-select --install
$ xcode-select --switch /Library/Developer/CommandLineTools
```

## Troubleshooting on macOS MacOS 10.14.2 (Mojave)

If you get an error like:

```
ld: library not found for -lssl (this usually means you need to install the development package for libssl)
```

> you may need to install OpenSSL and link pkg-config to OpenSSL

Reviewer 2: ```suggestion
you may need to install OpenSSL and link pkg-config to OpenSSL:
```

```
brew install openssl
```

```
export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:/usr/local/opt/openssl/lib/pkgconfig
```

> ```

Reviewer 2: This is not code.
```suggestion
```

> As with other keg-only formulas there are some caveats shown in brew info <formula> that advertise how to link pkg-config with this library.

Reviewer 2: ```suggestion
As with other keg-only formulas there are some caveats shown in `brew info <formula>` that shows how to link pkg-config with this library.
```

Reviewer 2: @dustinlahr was the commit batch feature not available? It allows you to squash all suggestions into one commit.

Author: I did not see it, so I just did it individually.

Reviewer 2: No worries!

> Crystal compiler by default will use pkg-config to find the location of libraries to link.

Reviewer 2: ```suggestion
The Crystal compiler will by default use pkg-config to find the locations of libraries to link with.
```

> ```

Reviewer 2: ```suggestion
```
